bmc_dump_entry.hpp

```diff
@@ -52,14 +54,15 @@ class Entry :
      */
     Entry(sdbusplus::bus::bus& bus, const std::string& objPath, uint32_t dumpId,
           uint64_t timeStamp, uint64_t fileSize,
-          const std::filesystem::path& file,
+          const std::filesystem::path& file, std::string genId,
           phosphor::dump::OperationStatus status,
```

Comment:
Is there any reason to add this new argument at the middle ? If this is added at the end, you can pass this as a default argument and assign that to null. But before taking this comment, please check if the default arguments are allowed at this repo.